implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -53,23 +53,20 @@ public JwtContext parse(final String token, final JWTAuthContextInfo authContext
             JwtConsumerBuilder builder = new JwtConsumerBuilder()
                     .setRequireExpirationTime();
 
-            if (authContextInfo.getWhitelistAlgorithms().isEmpty()) {
-                builder.setJwsAlgorithmConstraints(
-                        new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST,
-                                AlgorithmIdentifiers.RSA_USING_SHA256));
-            } else {
+            if (!authContextInfo.getWhitelistAlgorithms().isEmpty()) {
                 builder.setJwsAlgorithmConstraints(
                         new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST,
                                 authContextInfo.getWhitelistAlgorithms().toArray(new String[0])));
             }
-
             if (authContextInfo.isRequireIssuer()) {
                 builder.setExpectedIssuer(true, authContextInfo.getIssuedBy());
             } else {
                 builder.setExpectedIssuer(false, null);
             }
             if (authContextInfo.getSignerKey() != null) {
                 builder.setVerificationKey(authContextInfo.getSignerKey());
+            } else if (authContextInfo.getHmacSecret() != null) {
```

Comment:
This is something I'm not sure about. I doubt we should let the users configure the signing private keys directly in the application.properties. My only plan was to support a secret key location possibly restricting it to HTTPS based JWKs only for a start and may be relaxing it to support the file system.